NuKeeper/Engine/RepositoryUpdater.cs

```diff
@@ -108,6 +108,9 @@ private static void GitInit(IGitDriver git, RepositoryData repository)
         {
             git.Clone(repository.Pull.Uri);
             repository.DefaultBranch = repository.DefaultBranch ?? git.GetCurrentHead();
+            // checkout will change the branch to the correct one if it is changed
+            // with the --targetBranch option
+            git.Checkout(repository.DefaultBranch);
```

Comment:
What happens if we already checkout a branch? Is that Checkout done twice now?

Reply:
We should only do a checkout if our currentBranch != targetBranch right? So if i don't use the targetBranch option this will checkout the branch that is already checked out, so basically doing it twice?

Reply:
I will modify the clone to make add the branch info. That will fix that